### PR TITLE
ipq807x: yuncore ax880 add dualboot support

### PR DESCRIPTION
**Pull request by legale**
ipq807x: this pr corrects some shortcomings for the yuncore ax880 and add dualboot support for this device

- new env vars will be saved to the env part on sysupgrade via platform.sh script
- bootable slot is controller by env var owrt_active instead of stock fw smem controlled bootable slot. 
- uboot will increment bootcount env var on each boot, bootcount script will reset this counter on successfull and stable boot (5 min after boot)
- sysupgrade will flash new fw to the unused slot based on env var owrt_active

### CODE REVIEW

**Comment by Neustradamus** [comment]
To follow this PR

**Comment by Neustradamus** [comment]
@legale: Any progress on it?

**Comment by legale** [comment]
> @legale: Any progress on it?

So much time has passed. I've already forgotten about that PR. In fact, I set up a dual boot on my own system.
I'll gather my strength and perhaps push it again.

**Comment by Neustradamus** [comment]
@grosjo: Have you progressed on your PR?

**Comment by Neustradamus** [comment]
@legale: Have you progressed on your PR?

File: target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh [hunk 1]
```diff
@@ -4,6 +4,28 @@ REQUIRE_IMAGE_METADATA=1
...
+ax880_env_setup() {
+	local ubifile=$(board_name)
+	local active=$(fw_printenv -n owrt_slotactive)
+	[ -z "$active" ] && active=$(hexdump -s 0x94 -n 4 -e '4 "%d"' /dev/mtd$(find_mtd_index 0:bootconfig))
+	cat > /tmp/env_tmp << EOF
+owrt_slotactive=${active}
+owrt_bootcount=0
+bootfile=${ubifile}.ubi
+owrt_bootcountcheck=if test \$owrt_bootcount > 4; then run owrt_tftprecover; fi; if test \$owrt_bootcount = 3; then run owrt_slotswap; else echo bootcountcheck successfull; fi
+owrt_bootinc=if test \$owrt_bootcount < 5; then echo save env part; setexpr owrt_bootcount \${owrt_bootcount} + 1 && saveenv; else echo save env skipped; fi; echo current bootcount: \$owrt_bootcount
```
Comment (openwrt-ai) [inline target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh:16]: The `>` operator in U-Boot's `test` command is a string comparison, not numeric. In `cmd/test.c`, `>` maps to `OP_STR_GT` (`strcmp(...) > 0`); numeric greater-than is `-gt`. So `test $owrt_bootcount > 4` compares lexicographically: e.g. `"10" > "4"` is false (`'1' < '4'`), and even `"5" > "4"` works only by accident of single-digit ASCII ordering. Once `owrt_bootcount` reaches double digits the tftp-recover path never triggers.

```suggestion
owrt_bootcountcheck=if test \$owrt_bootcount -gt 4; then run owrt_tftprecover; fi; if test \$owrt_bootcount = 3; then run owrt_slotswap; else echo bootcountcheck successfull; fi
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh [hunk 2]
```diff
@@ -4,6 +4,28 @@ REQUIRE_IMAGE_METADATA=1
...
+bootcmd=run owrt_bootinc && run owrt_bootcountcheck && run owrt_slotselect && run owrt_bootlinux
```
Comment (openwrt-ai) [inline target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh:17]: Same U-Boot string-vs-numeric issue as in `owrt_bootcountcheck`: `test \$owrt_bootcount < 5` is a string comparison (`OP_STR_LT` in `cmd/test.c`), so the guard misbehaves for multi-digit values. Use the numeric operator `-lt`.

```suggestion
owrt_bootinc=if test \$owrt_bootcount -lt 5; then echo save env part; setexpr owrt_bootcount \${owrt_bootcount} + 1 && saveenv; else echo save env skipped; fi; echo current bootcount: \$owrt_bootcount
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh [hunk 3]
```diff
@@ -114,15 +136,18 @@ platform_do_upgrade() {
...
+		else
+			CI_UBIPART="rootfs_1"
 		fi
-		# force altbootcmd which handles partition change in u-boot
-		fw_setenv bootcount 3
-		fw_setenv upgrade_available 1
+		# reset bootcount
+		fw_setenv owrt_bootcount 0
+		#switch active fw slot
+		fw_setenv owrt_slotactive $((1 - active))
```
Comment (openwrt-ai) [inline target/linux/qualcommax/ipq807x/base-files/lib/upgrade/platform.sh:150]: If `owrt_slotactive` is unset/empty, `active` is empty and `$((1 - active))` expands to `$((1 - ))`, which is an arithmetic syntax error — `fw_setenv` then receives no value and the slot is not toggled, so the new firmware just flashed to `rootfs_1` would not be selected on next boot. The `[ "$active" = "1" ]` test above tolerates an empty `active` (falls through to `rootfs_1`), but this line does not. Consider defaulting, e.g. `active="${active:-0}"` after reading it, so both branches stay consistent.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by openwrt-ai — COMMENTED** [review]
Reviewed 1 new commit; flagged a U-Boot `test` string-vs-numeric comparison bug in the boot-count env script and an arithmetic edge case in the slot-toggle logic. See inline comments.

---
_Generated by [Claude Code](https://claude.ai/code)_
